client/profile-wizard/steps/business-details/flows/selective-bundle/selective-extensions-bundle/index.js

```diff
@@ -327,29 +373,64 @@ export const SelectiveExtensionsBundle = ( {
 	productTypes,
 } ) => {
 	const [ showExtensions, setShowExtensions ] = useState( false );
-	const [ values, setValues ] = useState( {} );
+	const [ values, setValues ] = useState( baseValues );
+	const [ installableExtensions, setInstallableExtensions ] = useState(
+		initialInstallableExtensions
+	);
+	const [ isFetching, setIsFetching ] = useState( true );
+
+	const allowMarketplaceSuggestions = useSelect( ( select ) =>
+		select( SETTINGS_STORE_NAME ).getSetting(
+			'wc_admin',
+			'allowMarketplaceSuggestions'
+		)
+	);
 
 	useEffect( () => {
-		const initialValues = installableExtensions.reduce(
-			( acc, curr ) => {
-				const plugins = getVisiblePlugins(
-					curr.plugins,
-					country,
-					industry,
-					productTypes
-				).reduce( ( pluginAcc, { slug } ) => {
-					return { ...pluginAcc, [ slug ]: true };
-				}, {} );
+		const setLocalInstallableExtensions = () => {
+			const initialValues = createInitialValues(
+				installableExtensionsData,
+				country,
+				industry,
+				productTypes
+			);
+			setInstallableExtensions( installableExtensionsData );
+			setValues( initialValues );
+			setIsFetching( false );
+		};
 
-				return {
-					...acc,
-					...plugins,
-				};
-			},
-			{ install_extensions: true }
-		);
-		setValues( initialValues );
-	}, [ country ] );
+		if (
+			window.wcAdminFeatures &&
+			window.wcAdminFeatures[ 'remote-extensions-list' ] === true &&
+			allowMarketplaceSuggestions
+		) {
+			apiFetch( {
+				path: '/wc-admin/onboarding/free-extensions',
+			} )
+				.then( ( results ) => {
+					const transformedExtensions = transformRemoteExtensions(
```

Comment:
In the remote payment methods PR, I parsed the locales on the server.  My thinking was that we could minimize the amount of data sent to and parsed by the client.

But curious to hear your thoughts on this if there are benefits to parsing client-side.

Reply:
Oh right, way better on the server. Done in https://github.com/woocommerce/woocommerce-admin/pull/6952/commits/919f3029ab0680fe589b38a3e4a44a81b5b54269